lib/rules/no-array-prototype-extensions.js

```diff
@@ -317,8 +317,10 @@ function applyFix(callExpressionNode, fixer, context, options = {}) {
       if (callArgs.length === 1) {
         return [
           fixer.insertTextBefore(
-            callExpressionNode,
-            `${sourceCode.getText(calleeObj)}[${sourceCode.getText(callArgs[0])}]`
+            callExpressionNode, 
+            callExpressionNode.callee.optional
+              ? `${sourceCode.getText(calleeObj)}?.[${sourceCode.getText(callArgs[0])}]` 
+              : `${sourceCode.getText(calleeObj)}[${sourceCode.getText(callArgs[0])}]`
```

Comment:
Can we put the ternary inside this statement instead of mostly duplicating the line?

```suggestion
            `${sourceCode.getText(calleeObj)}${callExpressionNode.callee.optional ? '?.' : ''}[${sourceCode.getText(callArgs[0])}]` 
```

Reply:
yeah, good idea, I will clean this part.